packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -186,7 +227,12 @@ export class Outbox {
 					contents: message.contents,
 					metadata: message.metadata,
 					compression: message.compression,
+					referenceSequenceNumber: message.referenceSequenceNumber,
 				})),
+				// Partial flushing will ensure that all messages in the same batch have the same
+				// reference sequence number. If disabled, messages may have different sequence numbers
+				// within the same batch, so the loader must rely on individual message reference sequence numbers.
+				this.params.config.disablePartialFlush ? undefined : batch.referenceSequenceNumber,
```

Comment:
I'd always use batch.referenceSequenceNumber here. We already (before this change) use single ref sequence number for a batch - DeltaManager does that, plus optinal compresison does it (on top of it).
I do not see much value in allowing to go back and re-implement per-message ref sequence number capability - I'd always pass batch.referenceSequenceNumber

Reply:
But if we `disablePartialFlush` is set to true, we won't be splitting the batch in case we find a different reference sequence number. 

This means that the batch will have different sequence numbers and we're going to be lying to the loader about the shape of the whole batch.

We can either:
- remove the `disablePartialFlush` feature gate
- keep this as is and if we supply `undefined` here, the delta manager should use it's own 'lastprocessedseqnum` instead (based on [the comment above](https://github.com/microsoft/FluidFramework/pull/14150#discussion_r1107604283))

Reply:
It's no different than what we have today. I.e. today, DeltaManager will overwrite ref seq number. If we always pass ref seq number (even if disablePartialFlush is true), then it's either noop, or we will supply smaller (earlier) ref seq number, but it will be more correct (assuming that we have newer loader that picks it up).
It's a small change in behavior, so yes, there is some risk, but I think it's tiny.

That said, I think it's beneficial (for time being) to use last op (in a batch) ref seq number when calculating ref sequence number for a batch, not first op. If they are different, it's less risk to use slightly higher number (but possibly lower than what DeltaManager might have stamped today)

Reply:
Thank you, it makes sense. I'll make the change regarding the batch reference sequence number. I agree it's less risk to use the highest ref seq num.